### PR TITLE
Http response thread pool

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- Thanks go to @bspark8 #6678 for testing and verifying the improvement of processing/sending http request on a different thread.
- This PR adds a thread pool to the http_plugin used for processing/sending http requests.
- Remove auto load of http_plugin in nodeos main.cpp since not all nodeos need the http_plugin and the extra threads it spawns.
- Application logic is still processed on the application thread.
- See #6678 for additional info

## Consensus Changes

None

## API Changes

None

## Documentation Additions

- http_plugin has new option:
  --http-threads arg (=2)               Number of worker threads in http thread pool

### CODE REVIEW

**Comment by heifner** [comment]
I missed that you are passing the new io_context to init. Thanks for pointing that out.

**Comment by heifner** [comment]
@bspark8 I've added a new io_context for the websocketpp server as you suggested.

**Comment by bspark8** [comment]
@heifner Thank you for additional threading for the part that receives the HTTP requests.
In addition, I did the same test that I did in the previous PR(#6678) in the current patch. Below is the result.

Current patch
- About 1137 TPS (22% performance improvement)
- Each thread CPU usage
main: 88%
receiving HTTP requests: 52%
responding to the HTTP requests: 14%

File: plugins/http_plugin/http_plugin.cpp [hunk 1]
```diff
@@ -138,6 +139,13 @@ namespace eosio {
 
          websocket_server_type    server;
 
+         uint16_t                                    thread_pool_size = 2;
```
Comment (brianjohnson5972) [inline plugins/http_plugin/http_plugin.cpp]: So we still want to default to 2 threads, even though we can have 1?

Comment (heifner) [reply plugins/http_plugin/http_plugin.cpp]: Since we now only load http_plugin if we need it, sure.

**Pull request merged**